flagsmith/flagsmith.py

```diff
@@ -208,6 +224,8 @@ def _get_identity_flags_from_document(
         self, identifier: str, traits: typing.Dict[str, typing.Any]
     ) -> Flags:
         identity_model = self._build_identity_model(identifier, **traits)
+        if not is_environment_model(self._environment):
+            raise TypeError("Environment should not be 'None'")
```

Comment:
Same idea. May be worth while to make a single private method to raise the type error if present. Not sure.

Reply:
Agreed, I will see if a `TypeGuard` is what I need here.